HPXMLtoOpenStudio/resources/defaults.rb

```diff
@@ -6111,102 +6124,133 @@ def self.get_water_heater_heating_capacity(fuel, nbeds, num_water_heaters, nbath
   # and bathrooms in the home.
   #
-  # Source: Table 8. Benchmark DHW Storage and Burner Capacity in 2014 BA HSP
-  #
   # @param fuel [String] Water heater fuel type (HPXML::FuelTypeXXX)
+  # @param is_hpwh [Boolean] True if a heat pump water heater
   # @param nbeds [Integer] Number of bedrooms in the dwelling unit
   # @param nbaths [Integer] Number of bathrooms in the dwelling unit
+  # @param n_occ [Double] Number of occupants in the dwelling unit
   # @return [Double] Water heater tank volume (gal)
-  def self.get_water_heater_tank_volume(fuel, nbeds, nbaths = nil)
+  def self.get_water_heater_tank_volume(fuel, is_hpwh, nbeds, nbaths, n_occ)
+    # FUTURE: Take into account usage multipliers
+    # FUTURE: Incorporate number of occupants for conventional elec/gas storage WHs.
+
     if nbaths.nil?
       nbaths = Defaults.get_num_bathrooms(nbeds)
     end
 
-    if fuel != HPXML::FuelTypeElectricity # Non-electric tank WHs
-      case nbeds
-      when 0, 1, 2
-        return 30.0
-      when 3
-        if nbaths <= 1.5
-          return 30.0
-        else
-          return 40.0
-        end
-      when 4
-        if nbaths <= 2.5
-          return 40.0
-        else
-          return 50.0
-        end
-      else
+    if is_hpwh && !n_occ.nil? # Heat pump water heater
+      # Source: Jeff Maguire recommendation for ResStock when num occupants is known
+      if (nbeds <= 2) && (n_occ <= 3)
+        # Up to 2 bedrooms AND up to 3 occupants
         return 50.0
+      elsif (nbeds >= 4) || (n_occ >= 5)
+        # 4 or more bedrooms OR 5 or more occupants
+        return 80.0
+      else
+        # (3 bedrooms AND up to 4 occupants) OR (up to 3 bedrooms AND 4 occupants)
+        return 66.0
       end
-    else
-      case nbeds
-      when 0, 1
-        return 30.0
-      when 2
-        if nbaths <= 1.5
-          return 30.0
-        else
-          return 40.0
-        end
-      when 3
-        if nbaths <= 1.5
-          return 40.0
-        else
-          return 50.0
-        end
-      when 4
-        if nbaths <= 2.5
-          return 50.0
+
+    elsif fuel == HPXML::FuelTypeElectricity # Electric tank WHs
+      # Source: Table 8. Benchmark DHW Storage and Burner Capacity in 2014 BA HSP
+      if nbeds <= 1
+        tank_vol = 30.0
+      elsif nbeds <= 2
+        tank_vol = (nbaths <= 1.5 ? 30.0 : 40.0)
+      elsif nbeds <= 3
+        tank_vol = (nbaths <= 1.5 ? 40.0 : 50.0)
+      elsif nbeds <= 4
+        tank_vol = (nbaths <= 2.5 ? 50.0 : 66.0)
+      elsif nbeds <= 5
+        tank_vol = 66.0
+      else
+        tank_vol = 80.0
+      end
+
+      if is_hpwh
+        # Bump up size (e.g., 30/40 => 50, 50 => 66, 66 => 80)
+        # Note: There are very few HPWHs < 50 or > 80 gal in AHRI
+        if tank_vol < 50
+          tank_vol = 50.0
+        elsif tank_vol < 66
+          tank_vol = 66.0
         else
-          return 66.0
+          tank_vol = 80.0
         end
```

Comment:
And here's the new algorithm for HPWHs when number of occupants is not provided. Same as ERWHs except we bump up the size.